hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/supervisor/process/ConsumerProcess.java

```diff
@@ -26,35 +28,36 @@
 
     private final Retransmitter retransmitter;
 
-    private final java.util.function.Consumer<SubscriptionName> shutdownCallback;
+    private final java.util.function.Consumer<Signal> shutdownCallback;
 
-    private volatile boolean running = true;
+    private boolean running = true;
```

Comment:
Are we absolutely certain we can remove volatile? I understand that the run method processes signals and can set the flag only from the existing thread, however this is potentially not future-proof. Maybe we should at least add a comment here, that the access is assumed to be from a single thread?